llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4885,8 +4872,248 @@ void LoopVectorizationCostModel::collectElementTypesForWidening() {
   }
 }
 
+/// Estimate the register usage for \p Plan and vectorization factors in \p VFs.
```

Comment:
Is it necessary to move this function? It makes it a lot more difficult to review. If it can't be avoided then perhaps it should be moved in a separate NFC commit within the same PR or done in a different PR?

Reply:
I had to move it when I was using `LoopVectorizationCostModel` to check for in loop reductions, but now that I'm using the `isInLoop()` function from the reduction recipe we don't need these changes. Reverted.

Reply:
Hmm, the function has still moved so it's still hard to see what's changed. If it's easier you could always abandon the commit history and start again?